frontend/src/Components/Login.js

```diff
@@ -0,0 +1,113 @@
+import React, { useState, useEffect } from "react"
+import { useSelector, useDispatch, batch } from "react-redux"
+import { useNavigate } from "react-router-dom"
+
+import { API_URL } from "../../src/utils/constants"
+import user from "../../src/reducers/user"
+
+const Login = () => {
+  const [username, setUsername] = useState("")
+  const [password, setPassword] = useState("")
+  const [mode, setMode] = useState("signup")
+
+  let errorMessage = useSelector((store) => store.user.error)
+
+  const accessToken = useSelector((store) => store.user.accessToken)
+
+  const dispatch = useDispatch()
+  const navigate = useNavigate()
+
+  useEffect(() => {
+    if (accessToken && mode === "signin") {
+      navigate("/")
+    }
+  }, [accessToken, navigate])
+
+  //This function posts the form to the database
+  const onFormSubmit = (e) => {
+    e.preventDefault()
+
+    const options = {
+      method: "POST",
+      headers: {
+        "Content-Type": "application/json",
+      },
+      body: JSON.stringify({ username, password }),
+    }
+
+    fetch(API_URL(mode), options)
+      .then((res) => res.json())
+      .then((data) => {
+        console.log(data.response)
+        if (data.success) {
+          batch(() => {
+            dispatch(user.actions.setUserId(data.response.userId))
+            dispatch(user.actions.setUsername(data.response.username))
+            dispatch(user.actions.setAccessToken(data.response.accessToken))
+            dispatch(user.actions.setError(null))
+            if (mode === "signup") {
+              alert(`Successfully signed up!`)
+              window.location.reload()
```

Comment:
We see that you make this functionality that after the user has signed up, you must after that also sign in to see the content. Maybe you could add that to the alert message that it is more clear for user.